examples/helm-install/main.tf

```diff
@@ -18,55 +18,141 @@ terraform {
       source  = "tehcyx/kind"
       version = ">= 0.4"
     }
+    kubernetes = {
+      source  = "hashicorp/kubernetes"
+      version = ">= 2.27"
+    }
     tls = {
       source  = "hashicorp/tls"
       version = ">= 4.0"
     }
   }
 }
 
-# ==========================================
+# =============================================================================================
 # Construct KinD cluster
-# ==========================================
+# =============================================================================================
 
 resource "kind_cluster" "this" {
   name = "flux-e2e"
 }
 
 # ==========================================
-# Add deploy key to GitHub repository
+# Initialise a Github project
 # ==========================================
 
+resource "github_repository" "this" {
+  name        = var.github_repository
+  description = var.github_repository
+  visibility  = "public"
+  auto_init   = true
+}
+
+resource "github_branch_default" "this" {
+  repository = github_repository.this.name
+  branch     = "main"
+}
+
+# =============================================================================================
+# Add deploy key to GitHub repository
+# =============================================================================================
+
 resource "tls_private_key" "flux" {
   algorithm   = "ECDSA"
   ecdsa_curve = "P256"
 }
 
 resource "github_repository_deploy_key" "this" {
   title      = "Flux"
-  repository = var.github_repository
+  repository = github_repository.this.name
   key        = tls_private_key.flux.public_key_openssh
   read_only  = "false"
 }
 
-# ===============================================
-# Bootstrap KinD cluster using flux helm chart
-# ===============================================
+# =============================================================================================
+# Bootstrap KinD cluster using flux2 helm chart. This chart simply installs flux2
+# Ref: https://artifacthub.io/packages/helm/fluxcd-community/flux2
+# =============================================================================================
 
-# Ref: https://github.com/fluxcd-community/helm-charts/tree/main/charts/flux2
-resource "helm_release" "this" {
+resource "kubernetes_namespace" "flux_system" {
+  metadata {
+    name = "flux-system"
+  }
+
+  lifecycle {
+    ignore_changes = [metadata]
+  }
+}
+
+resource "helm_release" "flux2" {
   repository = "https://fluxcd-community.github.io/helm-charts"
   chart      = "flux2"
   version    = "2.12.4"
 
-  name             = "flux2"
-  namespace        = "flux-system"
-  create_namespace = true
+  name      = "flux2"
+  namespace = "flux-system"
 
   # Allows us to upgrade the version of flux from another repo and terraform not roll it back.
   lifecycle {
     ignore_changes = [
       version
     ]
   }
```

Comment:
I don't think this is Ok, by doing this how will people upgrade with TF is the chart version is ignored?